pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -361,6 +384,52 @@ private boolean forceLog(BrokerResponse brokerResponse, long totalTimeMs) {
     return false;
   }
 
+
+  private void scoreResponse(BrokerResponse response, RequestStatistics stats) {
+
+    try {
+
+      if (response == BrokerResponseNative.EMPTY_RESULT || response == BrokerResponseNative.NO_TABLE_RESULT
+          || response.getNumServersResponded() == 0 || !response.getProcessingExceptions().isEmpty()) {
+        // error response
+        _brokerMetrics.addMeteredTableValue(stats.getTableName(), BrokerMeter.ERROR_RESPONSE, 1);
+        return;
+      }
+
+      if ((response.getNumServersQueried() > response.getNumServersResponded())
+          || stats.getFanoutType() == RequestStatistics.FanoutType.HYBRID
+          && response.getNumConsumingSegmentsQueried() <= 0 || response.isNumGroupsLimitReached()) {
```

Comment:
is it worth noting here in a comment that it is possible that due to helix timings, the externalview may contain (at least theoretically) no consuming segments, and we will count that as a partial response even though the others are in the process of getting created?
I am not sure if it is useful to add a log only for the case when we have realtime tables but queries 0 consuming segments.
Lastly, does response.getNumConsumingSegmentsQueried() work correctly with HLC?